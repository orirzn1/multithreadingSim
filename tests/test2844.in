L11
S8
O6
N8

T0
I@0x00000000
LOAD $2, $6, 0x1
SUB $1, $6, $4
LOAD $0, $7, 0x3
ADDI $2, $6, 0x2
LOAD $3, $3, 0x4
LOAD $1, $4, $2
SUB $2, $1, $2
STORE $7, $1, $4
SUB $6, $5, $2
HALT $0

T1
I@0x00000000
ADDI $2, $3, 0x3
STORE $5, $6, $2
STORE $1, $6, 0x3
SUBI $3, $1, 0x4
HALT $0

T2
I@0x00000000
ADDI $3, $2, 0x3
STORE $0, $4, $6
SUBI $5, $6, 0x3
SUB $2, $4, $4
ADDI $7, $1, 0x3
SUBI $0, $7, 0x3
LOAD $7, $3, $7
ADDI $6, $5, 0x2
HALT $0

T3
I@0x00000000
ADDI $7, $1, 0x3
SUB $5, $6, $0
LOAD $4, $7, 0x3
SUBI $2, $2, 0x4
STORE $4, $3, $7
SUB $7, $4, $7
ADDI $5, $4, 0x2
SUB $5, $5, $0
HALT $0

T4
I@0x00000000
ADDI $6, $0, 0x2
LOAD $7, $2, $6
STORE $4, $0, $7
SUB $7, $0, $2
LOAD $6, $6, 0x0
ADDI $4, $5, 0x2
HALT $0

T5
I@0x00000000
LOAD $3, $5, 0x0
LOAD $3, $5, $2
LOAD $3, $3, $0
SUBI $4, $3, 0x4
ADD $7, $3, $5
SUBI $7, $3, 0x1
SUB $2, $1, $2
ADDI $2, $7, 0x4
HALT $0

T6
I@0x00000000
ADD $0, $7, $1
LOAD $3, $2, $5
SUB $0, $7, $7
SUB $6, $6, $1
SUBI $3, $1, 0x4
HALT $0

T7
I@0x00000000
ADDI $0, $4, 0x2
ADDI $7, $7, 0x1
SUB $7, $5, $0
SUB $5, $1, $3
STORE $5, $6, 0x0
LOAD $6, $7, $5
SUB $4, $1, $1
ADD $6, $3, $5
SUBI $3, $7, 0x2
STORE $4, $4, 0x1
HALT $0

D@0x0
0x5
0x8
0x8
0xA
0x2
0x8
0x8
