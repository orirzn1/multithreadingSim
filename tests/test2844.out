
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0x8	R4 = 0x0	R5 = 0x0	R6 = 0x2	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x3	R3 = 0xFFFFFFFC	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x3	R4 = 0x0	R5 = 0xFFFFFFFD	R6 = 0xFFFFFFFF	R7 = 0x8
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0x0	R4 = 0x8	R5 = 0xA	R6 = 0x0	R7 = 0x5
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x2	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x3	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFF
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFC	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 7:
	R0 = 0x2	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFC	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFE
Blocked MT CPI for this program 3.575758

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0x8	R4 = 0x0	R5 = 0x0	R6 = 0x2	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x3	R3 = 0xFFFFFFFC	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x3	R4 = 0x0	R5 = 0xFFFFFFFD	R6 = 0xFFFFFFFF	R7 = 0x8
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0x0	R4 = 0x8	R5 = 0xA	R6 = 0x0	R7 = 0x5
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x2	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x3	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFF
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFC	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 7:
	R0 = 0x2	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFC	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFE
Finegrained Multithreading CPI for this program 1.106061

